### ISSUE TITLE
1.5: "oneOf" constraint for "tools" limits to either/or not both

### ISSUE THREAD

**Issue by mrutkows** [CycloneDX/specification#271]
Going line-by-line through the v1.5 schema and see the following use of the 
```
oneOf
```
 constraint (i.e., **“is valid if exactly one of the sub-schemas is valid”**)…
```
"tools": {
  "oneOf":
    "properties": {
      "components": {
        ...
      },
      "services": {
        ...
      }
    }
```
However, if my tooling is a hybrid of both “components” and “services” the `anyOf`  constraint (i.e.  “**valid if at least one of the sub-schemas is valid**”) would make more sense and not restrict me to either just "components" or "services"?

This especially makes sense where “tools” is reused for many purposes, such as CI/CD in workflows… where it is almost always hybrid.

Spec. Ref: https://json-schema.org/understanding-json-schema/reference/combining.html

**Comment by jkowalleck** [comment]
you are right, current XML and current JSON schema define `tools` as either a set of `component` or `service` .

**Comment by jkowalleck** [comment]
this allso would affect the XML - where it is a `choice` - the equivalent of a `oneOf`
